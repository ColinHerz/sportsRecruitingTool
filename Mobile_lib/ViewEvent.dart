import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'Session.dart';
import 'dart:async';

class ViewEvent extends StatefulWidget{
  final String eventID;

  ViewEvent(this.eventID);

  _ViewEvent createState() => _ViewEvent(eventID);
}

class _ViewEvent extends State<ViewEvent>{
  final String eventID;

  _ViewEvent(this.eventID);

  Future<void> getEvent() async {
    final http.Response _response = await http.get(
      'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/getEventResults/$eventID',
      headers: headers,
    );

    if (_response.statusCode == 200) {
      print(_response.body);
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }

  void initState(){
    super.initState();
    getEvent();
  }

  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('View Event'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Padding(
                padding: EdgeInsets.fromLTRB(0, 0, 0, 10),
                child: Text("Event")
            ),
            Text("Stuff"),
            RaisedButton(
                onPressed: (){
                  Navigator.pop(context);
                },
                child: Text('Back')
            ),
          ],
        ),
      ),
    );
  }
}
