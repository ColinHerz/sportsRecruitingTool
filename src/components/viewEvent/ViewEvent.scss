@import "../../colors";

#event {
	margin: auto;
	width: 60%;

	display: flex;
	flex-direction: column;
	justify-content: left;
	align-items: flex-start;

	background-color: $white;
	color: $black;

	font-family: "Open Sans";

	h2 {
		font-family: "Ubuntu";
		font-size: 48px;
	}

	h3 {
		font-family: "Open Sans";
		color: $gray;
		margin-top: -.4em;
	}

	#scores {
		width: 100%;
		text-align: left;
	}

	#scores h3 {
		color: $black;
	}

	span {
		color: $black;
		font-weight: 700;
	}

	ol li {
		font-size: 24px;
		float: left;
		margin: 0 12px;
		padding: 0px 20px 0px 0;
	}
	ol {
		margin-left: -25px;
	}
	p {
		font-weight: 400;
	}
}
