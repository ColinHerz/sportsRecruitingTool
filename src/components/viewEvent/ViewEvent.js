import apiCall from "../../api/api.js";
import { Link, useParams } from "react-router-dom";
import PropTypes from "prop-types";
import React, { useEffect, useState } from "react";

import "./ViewEvent.scss";

const ViewEvent = props => {
	const { eid } = useParams();

	const [firstLoad, setFirstLoad] = useState(true);

	const [eventInfo, setEventInfo] = useState({});

	const [showError, setShowError] = useState(false);

	useEffect(() => {
		if (firstLoad) {
			apiCall(
				{
					endpoint: `/golf/getEventResults/${eid}`,
					type: `GET`
				},
				data => {
					if (data.status !== 200) {
						setShowError(true);
						return;
					}

					const response = JSON.parse(data.response);
					const startDate = new Date(response.startDate);
					const endDate = new Date(response.endDate);

					response.startDate = startDate.toDateString();
					response.endDate = endDate.toDateString();

					setEventInfo(response);
				},
				() => {
					setShowError(true);
				}
			);

			setFirstLoad(false);
		}
	}, [firstLoad, eid]);

	return (
		<main id="view-event">
			{
				firstLoad ?
					<p>Loading event...</p>:
					<React.Fragment>
						<section id="meta-event-info">
							{
								showError ?
									<p id="error-message">Something went wrong. Please try again later.</p>:
									null
							}

							<h2>{eventInfo.eventName}</h2>

							<h3>{eventInfo.course}</h3>

							<p><span className="bolder">Start Date:</span> {eventInfo.startDate}</p>

							<p><span className="bolder">End Date:</span> {eventInfo.endDate}</p>
						</section>

						<Link id="profile-link" to="/profile/">Add a match from your profile.</Link>

						<section id="scores">
						</section>
					</React.Fragment>
			}

		</main>
	);
};

ViewEvent.propTypes = {
	user: PropTypes.object.isRequired
};

export default ViewEvent;
