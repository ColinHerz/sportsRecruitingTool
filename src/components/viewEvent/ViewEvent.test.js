/* eslint-env jest */
import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";

import ViewEvent from "./ViewEvent.js";

const genProps = () => {
	return {
		user: {}
	};
};

describe(`Rendering`, () => {
	it(`Renders beginning page correctly`, () => {
		const props = genProps();

		const component = renderer.create(
			<MemoryRouter>
				<ViewEvent {...props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
