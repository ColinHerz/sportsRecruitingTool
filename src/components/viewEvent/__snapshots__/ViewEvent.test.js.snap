// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders beginning page correctly 1`] = `
<main
  id="view-event"
>
  <p>
    Loading event...
  </p>
</main>
`;
